### PR TITLE
Add documentation for Playwright helpers

### PR DESCRIPTION
**Pull request by anagstef**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1529/testing/playwright

### This PR:

- Adds documentation for helper functions in the Playwright integration of the `@clerk/testing` package.

### CODE REVIEW

File: docs/testing/playwright.mdx [hunk 1]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
   ```
 </Steps>
 
+## Test Helpers
+
+The `@clerk/testing` package also provides some helper functions to sign in/sign out with Clerk in your Playwright tests without having to interact with the UI.
+To use these commands, you just have to import the `clerk` object from the `@clerk/testing/playwright` package.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:82]: Removing "just do X" to not make feel people who struggle with the instructions after it

```suggestion
To use these commands, import the `clerk` object from the `@clerk/testing/playwright` package.
```

File: docs/testing/playwright.mdx [hunk 2]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+### `clerk.signIn`
+
+The `clerk.signIn` function is used to sign in a user using Clerk. This helper supports only the following first factor strategies:
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:86]:
```suggestion
The `clerk.signIn()` function is used to sign in a user using Clerk. This helper supports only the following first factor strategies:
```

File: docs/testing/playwright.mdx [hunk 3]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+- Password
+- Phone code
+- Email code
+
+Multi-factor authentication is not supported.
+
+> [!NOTE]
+> This helper function internally uses the `setupClerkTestingToken()` helper, so you don't need to call it separately.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:95]:
```suggestion
> `clerk.signIn()` internally uses the `setupClerkTestingToken()` helper, so you don't need to call it separately.
```

To not repeat "helper"

File: docs/testing/playwright.mdx [hunk 4]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Prerequisites
+
+- Before calling this function, you must call `page.goto`.
+- Before using this function, navigate to a non-protected page that loads Clerk.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:100]:
```suggestion
- Navigate to a non-protected page that loads Clerk via `page.goto()`, for example the index page
```

File: docs/testing/playwright.mdx [hunk 5]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Parameters
+
+`clerk.signIn` accepts an object with the following properties:
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:104]:
```suggestion
`clerk.signIn()` accepts an object with the following properties:
```

File: docs/testing/playwright.mdx [hunk 6]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+  - `string`
+
+  The user's password. This is required only if the `strategy` is set to `'password'`.
+
+  ---
+
+  - `setupClerkTestingTokenOptions?`
+  - `Object`
+
+  Options to pass to the `setupClerkTestingToken` function. Optional.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:142]:
```suggestion
  Options to pass to `setupClerkTestingToken()`. Optional.
```

File: docs/testing/playwright.mdx [hunk 7]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+</Properties>
+
+#### Example
+
+The following example demonstrates how to use the `clerk.signIn` helper function in a test to sign in a user.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:147]:
```suggestion
The following example demonstrates how to use `clerk.signIn()` in a test to sign in a user.
```

File: docs/testing/playwright.mdx [hunk 8]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+    page,
+    signInParams: { strategy: 'phone_code', identifier: '+15555550100' },
+  })
+  await page.goto('/protected')
+})
+```
+
+### `clerk.signOut`
+
+The `clerk.signOut` function is used to sign out the current user using Clerk.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:164]:
```suggestion
`clerk.signOut()` is used to sign out the current user using Clerk.
```

File: docs/testing/playwright.mdx [hunk 9]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Prerequisites
+
+- Before calling this function, yu must call `page.goto`.
+- Before using this function, navigate to a page that loads Clerk.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:169]:
```suggestion
- Navigate to a non-protected page that loads Clerk via `page.goto()`, for example the index page
```

File: docs/testing/playwright.mdx [hunk 10]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Parameters
+
+`clerk.signOut` accepts an object with the following properties:
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:173]:
```suggestion
`clerk.signOut()` accepts an object with the following properties:
```

File: docs/testing/playwright.mdx [hunk 11]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+  - `signOutOptions.redirectUrl?`
+  - `string`
+
+  The redirect URL to navigate to after sign out is complete.
+</Properties>
+
+#### Example
+
+The following example demonstrates how to use the `clerk.signOut` helper in a test to sign a user out.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:198]:
```suggestion
The following example demonstrates how to use `clerk.signOut()` in a test to sign out a user.
```

File: docs/testing/playwright.mdx [hunk 12]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+  await page.goto('/protected')
+  await clerk.signOut({ page })
+  await page.goto('/protected')
+  // should redirect to sign in page
+})
+```
+
+### `clerk.loaded`
+
+The `clerk.loaded` function asserts that Clerk has been loaded.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:218]:
```suggestion
`clerk.loaded()` asserts that Clerk has been loaded.
```

File: docs/testing/playwright.mdx [hunk 13]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Prerequisites
+
+- Before calling this command, you must call `page.goto`.
+- Before using this command, navigate to a page that loads Clerk.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:223]:
```suggestion
- Navigate to a non-protected page that loads Clerk via `page.goto()`, for example the index page
```

File: docs/testing/playwright.mdx [hunk 14]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+#### Parameters
+
+`clerk.loaded` accepts an object with the following properties:
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:227]:
```suggestion
`clerk.loaded()` accepts an object with the following properties:
```

File: docs/testing/playwright.mdx [hunk 15]
```diff
@@ -76,4 +76,175 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+<Properties>
+  - `page`
+  - `Object`
+
+  The Playwright page object to use for the sign-in operation.
+</Properties>
+
+#### Example
+
+The following example demonstrates how to use the `clerk.loaded` helper in a test to assert that Clerk has been loaded.
```
Comment (LekoArts) [inline docs/testing/playwright.mdx:238]:
```suggestion
The following example demonstrates how to use `clerk.loaded()` in a test to assert that Clerk has been loaded.
```

File: docs/testing/playwright.mdx [hunk 16]
```diff
@@ -76,4 +76,172 @@ Playwright is a well-established JavaScript testing framework. This guide aims t
...
+
+  The user's password. This is required only if the `strategy` is set to `'password'`.
+
+  ---
+
+  - `setupClerkTestingTokenOptions?`
+  - `Object`
+
+  Options to pass to `setupClerkTestingToken()`. Optional.
+</Properties>
```
Comment (anagstef) [inline docs/testing/playwright.mdx:142]: How should we showcase the properties here to be easier to read? The object is the following:
```ts
type PlaywrightClerkSignInParams = {
  page: Page; // playwright object
  signInParams: {
      strategy: 'phone_code' | 'email_code' | 'password';
      password?: string;
      identifier: string;
  };
  setupClerkTestingTokenOptions?: {
      frontendApiUrl?: string;
  };
};
```

**Review by alexisintech — APPROVED** [review]

**Pull request merged**